runtime/src/main/java/com/redhat/insights/core/httpclient/InsightsJdkHttpClient.java

```diff
@@ -43,7 +39,19 @@ public InsightsJdkHttpClient(
     this.logger = logger;
     this.configuration = configuration;
     this.sslContextSupplier = sslContextSupplier;
-    this.useMTLS = configuration.getMaybeAuthToken().isEmpty();
+  }
+
+  public InsightsJdkHttpClient(InsightsLogger logger, InsightsConfiguration configuration) {
```

Comment:
I don't understand what this constructor is trying to 'solve'

Reply:
I'd rather have that default supplier provided explicitly, otherwise it might get used.

Reply:
The added constructor shall delegate to the full-arguments one, rather than doing `this.field` direct manipulations.

The supplier is only called for the mTLS case, so what's the point of passing a supplier to the default `SSLContext`? I don't see any application beyond shorter tests here.

Reply:
Let's go with a throwing supplier to make sure it's never called on the non-mTLS path.

Reply:
👍